addons/block_code/code_generation/blocks_catalog.gd

```diff
@@ -302,3 +302,18 @@ static func get_property_setter_block_definition(variable: VariableDefinition) -
 	var block_def := get_variable_setter_block_definition(variable)
 	block_def.description = "Set the %s property" % variable.var_name
 	return block_def
+
+
+static func get_resource_block_definition(file_name: String) -> BlockDefinition:
+	var block_def := BlockDefinition.new()
+
+	# Block Definition's name cannot work with '.'
+	block_def.name = &"get_resource_%s" % file_name.replace(".", "_")
```

Comment:
I wonder what happens if the file name has a whitespace in it. Is `file_name` here already escaping it?

Reply:
According to the discussion https://github.com/endlessm/godot-block-coding/pull/305#discussion_r1859424547, the block's name can be a fixed name, like `get_resource_file_path`